pkg/endpoint/policy.go

```diff
@@ -126,172 +100,217 @@ func (e *Endpoint) cleanUnusedRedirects(owner Owner, oldMap policy.L4PolicyMap,
 	}
 }
 
-// Must be called with global endpoint.Mutex held
-func (e *Endpoint) regenerateConsumable(owner Owner) (bool, error) {
-	c := e.Consumable
+func getLabelArrays(owner Owner, cache *policy.ConsumableCache, minID, maxID policy.NumericIdentity) (LabelArrays, error) {
+	labels := map[policy.NumericIdentity]labels.LabelArray{}
+	reservedIDs := cache.GetReservedIDs()
+	var idx policy.NumericIdentity
+	for _, idx = range reservedIDs {
+		lbls, err := owner.GetCachedLabelList(idx)
+		if err != nil {
+			return nil, err
+		}
+		// Skip currently unused IDs
+		if lbls == nil || len(lbls) == 0 {
+			continue
+		}
+		labels[idx] = lbls
+	}
 
+	for idx = minID; idx < maxID; idx++ {
+		lbls, err := owner.GetCachedLabelList(idx)
+		if err != nil {
+			return nil, err
+		}
+		// Skip currently unused IDs
+		if lbls == nil || len(lbls) == 0 {
+			continue
```

Comment:
Out of curiosity, was this condition ever reached?

Reply:
I don't know, maybe not likely for the reserved identities, but must be likely when looping from min to max ID as I don't think we compact the ID space after IDs become unused.